pkg/cloud-provider/cloudapi/azure/azure_account_mgmt.go

```diff
@@ -80,10 +79,6 @@ func compareAccountCredentials(accountName string, existing interface{}, new int
 
 // extractSecret extracts credentials from a Kubernetes secret.
 func extractSecret(c client.Client, s *v1alpha1.SecretReference) (*v1alpha1.AzureAccountCredential, error) {
-	if s == nil {
-		return nil, fmt.Errorf("secret reference not found")
```

Comment:
One thing here, we have a lot of error messages same in Azure and AWS here, like this `secret reference not found` , shall we add some suffix here to distinguish? maybe "secret reference not found in Azure" and secret reference not found in AWS" may be better?

Reply:
As discussed offline, the errors will also print out stacktraces, so its not really hard to identify whether aws or azure is having problem